### DISCUSSION TITLE
How to configure DCPSInfoRepo to use ip addresses instead of hostname

### DISCUSSION THREAD

**Discussion by jwilliams8067** [OpenDDS/OpenDDS#4286] [Q&A] [answered]
I am using an older version of OpenDDS version 3.7 and I am attempting to use DCPSInfoRepo  for discovery.
I have multiple network interface in the machine being used and need to use a specific interface to communicate with the DCPSInfoRepo  running on another machine.
I do not have a DNS server, just machine names and ip addresses in /etc/hosts

I can only seem to get it to work if the machine hostname is aliased to the ip address of the network interface I am attempting to use. normally the hostname is aliased to a different ip address,

Is there any way to force OpenDDS to use IP addresses instead of hostnames when communicating with the DCPSInfoRepo  ?
I cannot use RTPS discovery.

**Comment by mitza-oci** [comment, marked as answer]
This may be helpful https://theaceorb.com/support/faq.html#Errors-048

**Reply by jwilliams8067** [reply]
I was really hoping that would help but I am still having the same issue.
My DCPSInfoRepo is on machine with a 10.10.6.x address, the 2nd machine has multiple network interfaces but eth5 is on the 10.10.6.x subnet, the hostname on the second machine is aliased to a network interface with a 10.30.1.x address.

I get the following when I set the ORBDebugLevel to 10 on the DCPSInfoRepo


TAO (31802|3069467456) - GIOP_Message_Base::dump_msg, Transport[19] recv GIOP message v1.2, 484 data bytes, my endian, Type Request[8]
GIOP message - HEXDUMP 496 bytes
47 49 4f 50 01 02 01 00  e4 01 00 00 08 00 00 00   GIOP............
03 00 00 00 00 00 00 00  34 00 00 00 14 01 0f 00   ........4.......
4e 55 50 00 00 00 19 00  00 00 00 01 00 00 00 52   NUP............R
6f 6f 74 50 4f 41 00 49  6e 66 6f 52 65 70 6f 00   ootPOA.InfoRepo.
00 00 00 00 01 00 00 00  49 6e 66 6f 52 65 70 6f   ........InfoRepo
11 00 00 00 61 64 64 5f  73 75 62 73 63 72 69 70   ....add_subscrip
74 69 6f 6e 00 00 00 00  00 00 00 00 01 03 00 00   tion............
4b 00 00 00 01 03 00 00  00 00 04 00 00 00 00 06   K...............
00 00 01 c1 01 03 00 00  00 00 04 00 00 00 00 06   ................
00 00 01 45 26 00 00 00  49 44 4c 3a 4f 70 65 6e   ...E&...IDL:Open
44 44 53 2f 44 43 50 53  2f 44 61 74 61 52 65 61   DDS/DCPS/DataRea
64 65 72 52 65 6d 6f 74  65 3a 31 2e 30 00 00 00   derRemote:1.0...
01 00 00 00 00 00 00 00  6c 00 00 00 01 01 02 00   ........l.......
0c 00 00 00 31 30 2e 33  30 2e 31 2e 32 31 31 00   ....**10.30.1.211**.
98 cc 00 00 1b 00 00 00  14 01 0f 00 52 53 54 e7   ............RST.
d8 0d 65 76 cc 06 00 00  00 00 00 01 00 00 00 01   ..ev............
00 00 00 00 02 00 00 00  00 00 00 00 08 00 00 00   ................
01 00 00 00 00 4f 41 54  01 00 00 00 18 00 00 00   .....OAT........
01 00 00 00 01 00 01 00  01 00 00 00 01 00 01 05   ................
09 01 01 00 00 00 00 00  01 00 00 00 ff ff ff 7f   ................
ff ff ff 7f 00 00 00 00  00 00 00 00 00 00 00 00   ................
3c 00 00 00 00 00 00 00  00 00 00 00 ff ff ff 7f   <...............
ff ff ff 7f 00 00 00 00  00 00 00 00 01 00 00 00   ................
ff ff ff ff ff ff ff ff  ff ff ff ff 00 00 00 00   ................
00 00 00 00 00 00 00 00  00 00 00 00 ff ff ff 7f   ................
ff ff ff 7f ff ff ff 7f  ff ff ff 7f 01 00 00 00   ................
04 00 00 00 74 63 70 00  19 00 00 00 01 00 37 34   ....tcp.......74
11 00 00 00 67 64 70 70  31 2d 65 74 68 35 3a 34   ....**gdpp1-eth**5:4
32 35 38 39 00 00 00 00  00 00 00 00 00 00 00 00   2589............
00 00 00 00 00 00 00 00  01 00 00 00 01 00 00 00   ................
00 00 00 00 01 00 00 00  00 00 00 00 00 00 00 00   ................
TAO (31802|3069467456) - Stub::base_profiles, acquired profile lock this = 0x814af38
TAO (31802|3069467456) - Stub::base_profiles, acquired profile lock this = 0x814bac8
TAO (31802|3069467456) - Invocation_Adapter::invoke_i, making a TAO_CS_THRU_POA_STRATEGY invocation
TAO (31802|3069467456) - Invocation_Adapter::invoke_i, making a TAO_CS_REMOTE_STRATEGY invocation
TAO (31802|3069467456) - Transport_Cache_Manager_T::fill_set_i, current_size = 1, cache_maximum = 512
TAO (31802|3069467456) - Transport_Cache_Manager_T::purge, Cache size after purging is [1]
TAO (31802|3069467456) - IIOP_Connector::begin_connection, to <10.30.1.211:52376> which should block
TAO (31802|3069467456) - TAO_LF_CH_Event[0]::state_changed_i, state LFS_IDLE->LFS_CONNECTION_WAIT
TAO (31802|3069467456) - IIOP_Connection_Handler[135613928]::IIOP_Connection_Handler, this=0x8150d18
TAO (31802|3069467456) - Transport_Connector::wait_for_connection_completion, transport [135613928], Connection not complete.
TAO (31802|3069467456) - Cache_IntId_T::Cache_IntId_T, this=0xbfffc2a4 Transport[135613928] is not connected
TAO (31802|3069467456) - Cache_IntId_T::recycle_state, ENTRY_UNKNOWN->ENTRY_CONNECTING Transport[135613928] IntId=0xbfffc2a4
TAO (31802|3069467456) - Transport_Cache_Manager_T::bind_i, Transport[135613928] @ hash:index{169791083:0}
TAO (31802|3069467456) - Transport_Cache_Manager_T::bind_i: Success Transport[135613928] @ hash:index{169791083:0}. Cache size is [2]
TAO (31802|3069467456) - Transport_Connector::wait_for_connection_completion, going to wait for connection completion on transport[135613928]
TAO (31802|3069467456) - Leader_Follower[135613928]::wait_for_event, Becoming client leader.
TAO (31802|3069467456) - Leader_Follower[135613928]::wait_for_event, (leader) enter reactor event loop
TAO (31802|3069467456) - TAO_LF_CH_Event[135613928]::state_changed_i, state LFS_CONNECTION_WAIT->LFS_CONNECTION_CLOSED

**Reply by jwilliams8067** [reply]
It turns out that my problem was an empty -ORBEndpoints option that was causing the ORB to not use the correct network interface.
Th provided link answers my original question -ORBDottedDecimalAddresses 1 forces the use of the IP address inplace of the host name.

**Comment by jrw972** [comment]
> I cannot use RTPS discovery.

Can you elaborate on this?  We are considering retiring and/or replacing the InfoRepo in favor of RTPS discovery so any feedback on why you can't use RTPS discovery out of the box is useful.

**Reply by jwilliams8067** [reply]
In the past I have had performance issues with RTPS discovery in OpenDDS 3.7 on RHEL6 that made it unusable.  So I cannot use RTPS discovery in this environment.. In newer operating systems with newer versions of OpenDDS I am successfully using RTPS discovery.

**Reply by jrw972** [reply]
Thanks.
